clients/client/src/main/java/org/projectnessie/client/auth/BasicAuthenticationProvider.java

```diff
@@ -47,7 +47,14 @@ public String getAuthTypeValue() {
   @Override
   public HttpAuthentication build(Function<String, String> configSupplier) {
     String username = configSupplier.apply(CONF_NESSIE_USERNAME);
+    if (username == null) {
```

Comment:
I am wondering, does it make sense to check as well for the String emptiness as well as Null or only Null check suffice here?

Reply:
I'm not sure it's worth complicating the code for that. This `if` is just for backward compatibility with existing clients. I imagine we're going to remove it once all clients migrate to Nessie 1.0+.